Source/DafnyPipeline.Test/FormatterIssues.cs

```diff
@@ -4,6 +4,83 @@ namespace DafnyPipeline.Test;
 
 [Collection("Singleton Test Collection - FormatterForTopLevelDeclarations")]
 public class FormatterIssues : FormatterBaseTest {
+  [Fact]
+  public void GitIssue3912FormatterCollectionArrow() {
+    FormatterWorksFor(@"
+const i :=
+  a in
+    B + // Previously it was not indented
+    C +
+    D &&
+  b in
+    B +
+    C
+");
+  }
+
+  [Fact]
+  public void GitIssue3912FormatterCollectionArrowA() {
+    FormatterWorksFor(@"
+const newline :=
+  set
+    i <-
+      PartA +
+      PartB +
+      PartC,
+    j <-
+      PartD
+    ::
+      f(i,j)
+
+const sameline :=
+  set i <-
+        PartA +
+        PartB +
+        PartC,
+    j <-
+      PartD
+    ::
+      f(i,j)
+
+");
+  }
+
+  [Fact]
+  public void GitIssue3912FormatterCollectionArrowB() {
+    FormatterWorksFor(@"
+const newlineOp :=
+  set
+    i
+    <-
+      PartA +
+      PartB +
+      PartC,
+    j
+    <-
+      PartD
+    ::
+      f(i,j)
+
+");
+  }
+
+  [Fact]
+  public void GitIssue3912FormatterCollectionArrowC() {
+    FormatterWorksFor(@"
+const sameLineOp :=
+  set i
+    <-
+      PartA +
+      PartB +
+      PartC,
+    j
+    <- PartD +
+       PartE
```

Comment:
Is this `PartE` intended to be aligned with `PartD`, or is it supposed to be one 2-space indent past the corresponding `<-`?

Reply:
It is intended to be aligned with PartD. Specifically, the "<-" is treated like an "in" binary operator, and I put the rule that if the "in" operator is not followed by a newline, then it starts [a new aligning indentation](https://github.com/dafny-lang/dafny/pull/3916/files#diff-99ac3ad316fe9cac2a05db0a8db42d2b1fe665fb21403fb25b02e50916005c3fR2350). If it is followed by a newline, then [it's indented by two spaces as usual](https://github.com/dafny-lang/dafny/pull/3916/files#diff-99ac3ad316fe9cac2a05db0a8db42d2b1fe665fb21403fb25b02e50916005c3fR2348). This matches usages of newlines I've seen for the second argument of "in" and "<-"